### PR TITLE
Ensure logging is configured for CLI commands

### LINKED ISSUE #27521: [CI] Vagrant tests fail with logging error
See eg. https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+packaging-tests/2175/console.

```
not ok 69 [TAR PLUGINS] install jvm-example with different logging modes and check output
not ok 69 [TAR PLUGINS] install jvm-example with different logging modes and check output
```

This started failing after core-cli was made its own module #27114, so it might be related? cc @hub-cap

### PR DESCRIPTION
**Pull request by jasontedor**
Any CLI commands that depend on core Elasticsearch might touch classes (directly or indirectly) that depends on logging. If they do this and logging is not configured, Log4j will dump status error messages to the console. As such, we need to ensure that any such CLI command configures logging (with a trivial configuration that dumps log messages to the console). Previously we did this in the base CLI command but with the refactoring of this class out of core Elasticsearch, we no longer configure logging there (since we did not want this class to depend on settings and logging). However, this meant for some CLI commands (like the plugin CLI) we were no longer configuring logging. This commit adds base classes between the low-level command and multi-command classes that ensure that logging is configured. Any CLI command that depends on core Elasticsearch should use this infrastructure to ensure logging is configured. There is one exception to this: Elasticsearch itself because it takes reponsibility into its own hands for configuring logging from Elasticsearch settings and log4j2.properties. We preserve this special status.

Closes #27521

### CODE REVIEW

File: core/src/main/java/org/elasticsearch/cli/CommandLoggingConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Holder class for method to configure logging without Elasticsearch configuration files for use in CLI tools that will not read such
+ * files.
+ */
+final class CommandLoggingConfigurator {
+
+    /**
+     * Configures logging without Elasticsearch configuration files based on the system property "es.logger.level" only. As such, any
+     * logging will written to the console.
```
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/cli/CommandLoggingConfigurator.java]: will _be_ written?

**Review by tlrx — APPROVED** [review]
LGTM, thanks Jason

File: core/src/main/java/org/elasticsearch/cli/CommandLoggingConfigurator.java (hunk 1 above)
Comment (jasontedor) [reply core/src/main/java/org/elasticsearch/cli/CommandLoggingConfigurator.java]: Thanks. I pushed 31477c20239a0afd1fd609b0719a31b18ef81d16.

**Comment by jasontedor** [comment]
Thanks @tlrx.

**Pull request merged**